src/Microsoft.AspNet.Mvc.WebApiCompatShim/HttpRequestMessage/HttpRequestMessageExtensions.cs

```diff
@@ -331,7 +331,10 @@ public static HttpResponseMessage CreateResponse<T>(this HttpRequestMessage requ
             if (result?.Formatter == null)
             {
                 // Return a 406 when we're actually performing conneg and it fails to find a formatter.
-                return request.CreateResponse(HttpStatusCode.NotAcceptable);
+                return new HttpResponseMessage(HttpStatusCode.NotAcceptable)
```

Comment:
cc @rynowak - the `request.CreateResponse` doesn't exist in CoreCLR and it ended up recursively calling this method.